### DISCUSSION TITLE
How can I run a Python Code or program in atom?

### DISCUSSION THREAD

**Discussion by Yuta0205** [atom/atom#23135] [Q&A]
Today, I wrote my first Python code, but I can't run it. When I go to file explorer and double click my python file, the python terminal window opens but disappears immediately, and nothing happens. Please tell me how to run a python code I made in Atom. (I'm on Windows 10 Home)

**Comment by icecream17** [comment]
There might be an error, which crashes the program causing it to exit immediately.
Or maybe your program is so fast that it finishes in 0 seconds.

You can use `try` `catch` to check if there's an error
And you can run `input("Press enter to finish")` (or in python 2, raw_input) to prevent the program from exiting

This is the simplest solution but the best solution is to run your program in a terminal, like `cmd`. In `cmd`, `cd <path to your file>`, and try `python3 <filename>`. You might need to search google or stackoverflow if this errors though.

That way when the program exits, it'll just return back to the cmd and you get to still see your text: (Note: I have the package pretty-errors, but that doesn't matter for this concept)
![image](https://user-images.githubusercontent.com/58114641/138537367-f5601b8a-d987-4203-ae30-ed5ec5a9068d.png)

**Comment by Danfro** [comment]
Although I haven't tried it myself yet, the [package "script"](https://github.com/atom-community/atom-script) seems to provide a tool for running scripts right in atom.

**Comment by Cjwe20** [comment]
> Although I haven't tried it myself yet, the [package "script"](https://github.com/atom-community/atom-script) seems to provide a tool for running scripts right in atom.
